js/background.js

```diff
@@ -483,40 +499,45 @@
         });
     }
 
-    function onMessageReceived(ev) {
-        var data = ev.data;
-        if (data.message.flags & textsecure.protobuf.DataMessage.Flags.PROFILE_KEY_UPDATE) {
-            var profileKey = data.message.profileKey.toArrayBuffer();
-            return ConversationController.getOrCreateAndWait(data.source, 'private').then(function(sender) {
-              return sender.setProfileKey(profileKey).then(ev.confirm);
-            });
-        }
-        var message = initIncomingMessage(data);
-
-        return isMessageDuplicate(message).then(function(isDuplicate) {
-            if (isDuplicate) {
-                console.log('Received duplicate message', message.idForLogging());
-                ev.confirm();
-                return;
-            }
+  /* eslint-enable */
+  /* jshint ignore:start */
+  async function onMessageReceived(ev) {
+    const { data } = ev;
+
+    // eslint-disable-next-line no-bitwise
+    if (data.message.flags & textsecure.protobuf.DataMessage.Flags.PROFILE_KEY_UPDATE) {
+      const profileKey = data.message.profileKey.toArrayBuffer();
+      const sender = ConversationController.getOrCreateAndWait(data.source, 'private');
+      await sender.setProfileKey(profileKey);
+      // TODO: Is `ev.confirm` a `Promise`? Original code returned it:
+      return ev.confirm();
+    }
 
-            var type, id;
-            if (data.message.group) {
-                type = 'group';
-                id = data.message.group.id;
-            } else {
-                type = 'private';
-                id = data.source;
-            }
+    const message = initIncomingMessage(data);
+    const isDuplicate = await isMessageDuplicate(message);
 
-            return ConversationController.getOrCreateAndWait(id, type).then(function() {
-                return message.handleDataMessage(data.message, ev.confirm, {
-                    initialLoadComplete: initialLoadComplete
-                });
-            });
-        });
+    if (isDuplicate) {
+      console.log('Received duplicate message', message.idForLogging());
+      // TODO: Is `ev.confirm` a `Promise`? Original code didn’t return it:
+      return ev.confirm();
     }
 
+    const { type, id } = data.message.group ?
+      { type: 'group', id: data.message.group.id } :
+      { type: 'private', id: data.source };
+
+    const processedMessage = await Message.process(data.message);
+
+    await ConversationController.getOrCreateAndWait(id, type);
+    return message.handleDataMessage(
+      processedMessage,
+      ev.confirm,
+      { initialLoadComplete }
+    );
+  }
+  /* jshint ignore:end */
+  /* eslint-disable */
+
     function onSentMessage(ev) {
```

Comment:
Can I just say that I feel dirty with eslint, jshint and different indentation right next to each other in the same file? I guess I always thought we'd update the entire file at once, but I suppose this is an easy way to get started...

Reply:
Agreed, it’s not ideal, but I felt it was more pragmatic than addressing 900+ lint errors in this file and introducing a lot of diff noise. Would be happy to do it in a separate PR but I feel doing these things incrementally might reduce the potential for bugs that could be introduced 😄